mlrun/projects/project.py

```diff
@@ -3040,7 +3040,7 @@ def run(
                 "Remote repo is not defined, use .create_remote() + push()"
             )
 
-        if engine not in ["remote"]:
+        if schedule or engine not in ["remote"]:
             # for remote runs we don't require the functions to be synced as they can be loaded dynamically during run
```

Comment:
why is it true for schedule as well?

Reply:
schedules use remote engine. a few lines below there is the engine override if a schedule was given